### PR TITLE
Add specification for DLT_LIN

### PR DESCRIPTION
**Pull request by kayoub5**

### CODE REVIEW

**Comment by guyharris** [comment]
So do you know whether this matches what Hannes Kaelber (hannes.kaelber@x2e.de, at the time he requested DLT_LIN) wanted when he requested DLT_LIN in [this message](https://seclists.org/tcpdump/2008/q1/191)?

**Comment by guyharris** [comment]
> So do you know whether this matches what Hannes Kaelber ([hannes.kaelber@x2e.de](mailto:hannes.kaelber@x2e.de), at the time he requested DLT_LIN) wanted when he requested DLT_LIN in [this message](https://seclists.org/tcpdump/2008/q1/191)?

Can I assume from "This document is based on email conversation with Hannes Kaelber" that the answer is "yes"?

**Comment by guyharris** [comment]
So that translates to this sequence of octets:

```
      8     7     6     5        |      4     3     2     1       
+--------------------------------+--------------------------------+
|                         msg_format_rev                          |
+-----------------------------------------------------------------+
|                            reserved                             |
+-----------------------------------------------------------------+
|                            reserved                             |
+-----------------------------------------------------------------+
|                            reserved                             |
+---------------------------------+-------------------------------+
|         payload_length          | message_type  | checksum_type |
+---------------------------------+---------------+---------------+
|                               PID                               |
+-----------------------------------------------------------------+
|                            checksum                             |
+-----------------------------------------------------------------+
|                             errors                              |
+-----------------------------------------------------------------+
|                             payload                             |
.                                                                 .
.                                                                 .
.                                                                 .
```

with the payload being an ISO 17987-2 N_PDU if message_type is 0 and with it being a 4-octet LIN Event type if message_type is 3?

Is the LIN Event type big-endian or little-endian?

**Comment by guyharris** [comment]
What is the interpretation of the msg_format_rev field (a single integral value?), what values are supported, and what value corresponds to this format?

For the checksum_type field, what does "unknown/error" indicate?  Presumably the "classic" and "enhanced" checksum are as described in section 5.2.2.7 "Checksum" of ISO 17987-3:2016.

Presumably the PID field is the protected identifier field preceding the first data field of the frame, as described in section 5.2.2.5 "PID field" of ISO 17987-3:2016.

What types of error do the flags in the error field indicate?:

- what is an overflow error?
- a checksum error is presumably an invalid checksum
- a parity error is presumably an error in the parity of the protected identifier field
- is a framing error a low-level transmissions/reception error?
- is a no slave response error a master request frame not getting a slave response frame?

**Comment by kayoub5** [comment]
@guyharris Yes, this document as a copy of an email from Hannes Kaelber (the email is a translation of a German documentation he used), as far as your questions goes, I will have to re-contact him. Would you like that I put you email in CC?

**Comment by guyharris** [comment]
> Would you like that I put you email in CC?

Yes, please.

**Comment by kayoub5** [comment]
@guyharris I don't see an email on your profile

**Comment by kayoub5** [comment]
@guyharris Hannes Kaelber is in vacation until mid April, I will re-contact him when he is back.

**Comment by kayoub5** [comment]
@guyharris document updated with Hannes Kälber feedback, good to merge?

**Comment by infrastation** [comment]
Please rebase on the current master branch if you can.

**Comment by kayoub5** [comment]
@infrastation done

File: style.css [hunk 1]
```diff
@@ -258,3 +258,13 @@ img.cibadge {
 	margin-top: 1px;
 	margin-bottom: 1px;
 }
+
```
Comment (mcr) [inline style.css:261]: I guess I'm surprised that we need new CSS, but hey if it makes this prettier... great!

**Review by mcr — APPROVED** [review]

**Comment by kayoub5** [comment]
Anything pending for this PR to be merged?

**Pull request merged**